torch/_subclasses/functional_tensor.py

```diff
@@ -306,8 +316,23 @@ def unwrap(x):
             try:
                 # By default for python functionalization (for AOTAutograd), we reapply views.
                 old_apply_views = torch._functionalize_enable_reapply_views(True)  # type: ignore[attr-defined]
-                outs_unwrapped = func(*args_unwrapped, **kwargs_unwrapped)
-                outs_wrapped = pytree.tree_map_only(torch.Tensor, wrap, outs_unwrapped)
+
+                # Sometimes these functions cannot be directly dispatched to functionalize key
+                # because args are sometimes not functional tensors for some reason?
+                if func in FunctionalTensor.metadata_fns:
+                    outs_unwrapped = func(*args_unwrapped, **kwargs_unwrapped)
+                    outs_wrapped = pytree.tree_map_only(
+                        torch.Tensor, wrap, outs_unwrapped
+                    )
+                else:
+                    outs_unwrapped = func._op_dk(
```

Comment:
nit: add a comment explaining why we need to use `func._op_dk` instead of directly calling `func` (answer: when we dispatch to the C++ functionalization kernel, we might need to jump back to the PreDispatch mode stack afterwards, to handle any other PreDispatch modes underneath FunctionalTensorMode. If we call `func()` directly, we would need to exclude `PreDispatch` from the TLS in order to avoid infinite looping, but this would prevent us from coming back to PreDispatch later)